client_interface.py

```diff
@@ -15,11 +15,8 @@
 class Interface:
     def __init__(self):
         # Game attributes
-        self.window = create_window()
-        self.window.push_handlers(
-            on_draw=self.window_draw,
-            on_text=self.on_text, )
-        self.state = InterfaceState()
+        self.window = create_window(self.window_draw, self.on_text)
+        self.interface_state = InterfaceState(callback=self.send_state_to_server)
```

Comment:
Tady klient říká: Milý nový InterfaceState, až se něco změní, zavolej tuhletu funkci!
(U klienta se jmenuje `self.send_state_to_server`, ale klidně by to mohl být třeba `print`; InterfaceStatu je jedno co bude po každé změně volat.)